Module.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LinkedDataSets;
+
+use BulkExport\Job\Export as JobExport;
+use LinkedDataSets\Application\Job\CatalogDumpJob;
+use LinkedDataSets\Domain\Job\CreateCatalogDumpJob;
+use Laminas\EventManager\Event;
+use Laminas\EventManager\SharedEventManagerInterface;
+use Laminas\ServiceManager\ServiceLocatorInterface;
+use Omeka\Api\Adapter\ItemAdapter;
+use Omeka\Api\Manager;
+use Omeka\Api\Representation\ItemRepresentation;
+use Omeka\Api\Representation\ResourceClassRepresentation;
+use Omeka\Api\Request;
+use Omeka\Job\Dispatcher;
+use Omeka\Module\AbstractModule;
+use Omeka\Module\Exception\ModuleCannotInstallException;
+use Omeka\Module\Module as DefaultModule;
+use Omeka\Stdlib\Message;
+use Omeka\Api\Exception\ValidationException;
+
+final class Module extends AbstractModule
+{
+    private ?Dispatcher $dispatcher = null;
+    private $api = null;
+    private const MODULE_DEPENDENCIES = [
+        ['name' => 'CustomVocab', 'version' => '1.7.1'],
+        ['name' => 'AdvancedResourceTemplate', 'version' => '3.4.4.17'],
+        ['name' => 'NumericDataTypes', 'version' => '1.11.0'],
+    ];
+
+    private const FOLDERS = [
+        ['path' => 'files/datacatalogs/'],
+        ['path' => 'files/datadumps/'],
+    ];
+
+    public function getConfig()
+    {
+        return include __DIR__ . '/config/module.config.php';
+    }
+
+    public function setServiceLocator(ServiceLocatorInterface $serviceLocator)
+    {
+        $this->serviceLocator = $serviceLocator;
+    }
+
+    public function install(ServiceLocatorInterface $serviceLocator)
+    {
+
+//        $this->checkPrerequisites($serviceLocator);
+//        $this->createFoldersIfTheyDontExist();
+//        $this->installSchemaOrgVocabulary($serviceLocator);
+//        $this->installCustomVocabularies();
+//        $this->installTemplates();
+    }
+
+    protected function checkPrerequisites(ServiceLocatorInterface $serviceLocator): void
+    {
+        /* @var DefaultModule $module */
+        foreach (self::MODULE_DEPENDENCIES as $moduleDependency) {
+            $module = $serviceLocator->get('Omeka\ModuleManager')
+                ->getModule($moduleDependency['name'])
+            ;
+
+            if ($module && version_compare($module->getIni('version') ?? '', $moduleDependency['version'], '<')) {
+                $translator = $serviceLocator->get('MvcTranslator');
+                $message = new Message(
+                    $translator->translate('This module requires the module "%s", version %s or above.'), // @translate
+                    $moduleDependency['name'], $moduleDependency['version']
+                );
+                throw new ModuleCannotInstallException((string) $message);
+            }
```

Comment:
1. Can you explain the purpose of this check? As I see it, it checks if the configured module version (from module.ini) matches the one set in the Module requirements class constant. I would expect you would use the ModuleManager here to load the required modules and test if they have the minimum required version you configured in the class constants.

2. Consider moving the minimum required versions of the module dependencies to the modules.ini. I feel that is a more natural place for config like this to hang out. But correct me if I'm wrong!

Reply:
It checks the version from the dependant installed module.ini to the required version in the constant of the consuming module. Not yet familiar with the usage of the consuming module.ini how to do that.

Reply:
About 2. That would really be useful. I am going to find out how the module.ini is read.

Reply:
Done this in json as ini isn't capable of multilevel arrays